packages/slice-machine/src/modules/useSliceMachineActions.ts

```diff
@@ -0,0 +1,66 @@
+import { useDispatch } from "react-redux";
+import { LoadingKeysEnum } from "@src/modules/loading/types";
+import { ModalKeysEnum } from "@src/modules/modal/types";
+import { modalCloseCreator, modalOpenCreator } from "@src/modules/modal";
+import {
+  startLoadingActionCreator,
+  stopLoadingActionCreator,
+} from "@src/modules/loading";
+import {
+  finishOnboardingCreator,
+  sendAReviewCreator,
+  skipReviewCreator,
+} from "@src/modules/userContext";
+import { getEnvironmentCreator } from "@src/modules/environment";
+import { ServerState } from "@models/server/ServerState";
+
+const useSliceMachineActions = () => {
+  const dispatch = useDispatch();
+
+  // Modal store
+  const closeLoginModal = () =>
+    dispatch(modalCloseCreator({ modalKey: ModalKeysEnum.LOGIN }));
+  const openLoginModal = () =>
+    dispatch(modalOpenCreator({ modalKey: ModalKeysEnum.LOGIN }));
+
+  // Loading store
+  const startLoadingReview = () =>
+    dispatch(startLoadingActionCreator({ loadingKey: LoadingKeysEnum.REVIEW }));
+  const stopLoadingReview = () =>
+    dispatch(stopLoadingActionCreator({ loadingKey: LoadingKeysEnum.REVIEW }));
+  const startLoadingLogin = () =>
+    dispatch(startLoadingActionCreator({ loadingKey: LoadingKeysEnum.LOGIN }));
+  const stopLoadingLogin = () =>
+    dispatch(stopLoadingActionCreator({ loadingKey: LoadingKeysEnum.LOGIN }));
+
+  // UserContext Store
+  const skipReview = () => dispatch(skipReviewCreator());
+  const sendAReview = () => dispatch(sendAReviewCreator());
+  const finishOnboarding = () => dispatch(finishOnboardingCreator());
+  const getEnvironment = (serverState: ServerState | undefined) => {
+    if (!serverState) return;
```

Comment:
I think the ServerState should not be undefined here but the function should not be called if it is underfined.

Reply:
Mistake fixed :)

Reply:
I fix that